### ISSUE TITLE
Form-o-fill doesn't extract properly fields/buttons with no 'name' attribute

### ISSUE THREAD

**Issue by kalpitk** [form-o-fill/form-o-fill-chrome-extension#92]
Here's an example website- http://www.istqb.org/login.html
Form-o-fill extracts the button as 
```
 {
    "selector": "input[name='']",
    "value": ""
  }
```
Which will give an error while execution.
![image](https://user-images.githubusercontent.com/29591472/27369432-7dc6ea08-5675-11e7-9a94-153681bf61a4.png)

Instead it should have been extracted as-
```
 {
    "selector": "button[type='submit']",
    "value": ""
  }
```

**This wrong extraction takes place, everytime when the field/button doesn't have 'name' attribute defined.**

**Comment by thenoseman** [comment]
Hi,

thanks for the issue report!

Indeed the extract functionality could need some work.

I'll take this on next.

**Comment by thenoseman** [comment]
Ideas:

Always check to see if "unique" already reached.

- Has id => use #id 
- Has class? use nodeName+id
- Unique element ? Use tag
- Not unique -> add parent in front and recurse.

**Comment by thenoseman** [comment]
Will be released with the next version (3.6.0)
